packages/ckeditor5-watchdog/src/editorwatchdog.ts

```diff
@@ -453,6 +497,27 @@ class EditorWatchdogInitPlugin {
 			} );
 		} );
 	}
+
+	/**
+	 * Restores the editor collaboration data - comment threads and suggestions.
+	 */
+	private _restoreCollaborationData() {
```

Comment:
We need to make sure that we don't add the same comment or suggestion twice. How it can happen? If there were starting comments/suggestions added in the integration plugin, like we describe in guides (look at load&save integration specifically).

Reply:
So, we need to either listen to which comments/suggestion were added, or we need save which were initially added, or here we need to skip these, which already exists (they were added earlier in `init()` methods of the integration plugins).

Reply:
It caused errors. Fixed in the newest commit

Reply:
Now as I think about it... I think that we actually should overwrite the already existing comment threads and suggestions as the saved data may be more up-to-date than the initial data (e.g. comments has been added to the thread).

Reply:
We also should remove threads and suggestions if needed.